pkg/resources/resource_source_postgres.go

```diff
@@ -38,42 +39,50 @@ var sourcePostgresSchema = map[string]*schema.Schema{
 		Optional:    true,
 	},
 	"table": {
-		Description: "Creates subsources for specific tables. If neither table or schema is specified, will default to ALL TABLES",
-		Type:        schema.TypeList,
+		Description: "Creates subsources for specific tables in the Postgres connection.",
+		Type:        schema.TypeSet,
 		Elem: &schema.Resource{
 			Schema: map[string]*schema.Schema{
-				"name": {
-					Description: "The name of the table.",
+				"upstream_name": {
+					Description: "The name of the table in the upstream Postgres database.",
 					Type:        schema.TypeString,
 					Required:    true,
 				},
-				"alias": {
-					Description: "The alias of the table.",
+				"upstream_schema_name": {
+					Description: "The schema of the table in the upstream Postgres database.",
+					Type:        schema.TypeString,
+					Optional:    true,
+					Computed:    true,
+				},
+				"name": {
+					Description: "The name of the table in Materialize.",
+					Type:        schema.TypeString,
+					Optional:    true,
+					Computed:    true,
+				},
+				"schema_name": {
+					Description: "The schema of the table in Materialize.",
+					Type:        schema.TypeString,
+					Optional:    true,
+					Computed:    true,
+				},
+				"database_name": {
+					Description: "The database of the table in Materialize.",
 					Type:        schema.TypeString,
 					Optional:    true,
+					Computed:    true,
 				},
 			},
 		},
-		Optional:      true,
-		MinItems:      1,
-		ConflictsWith: []string{"schema"},
-	},
-	"schema": {
-		Description:   "Creates subsources for specific schemas. If neither table or schema is specified, will default to ALL TABLES",
-		Type:          schema.TypeList,
-		Elem:          &schema.Schema{Type: schema.TypeString},
-		Optional:      true,
-		ForceNew:      true,
-		MinItems:      1,
-		ConflictsWith: []string{"table"},
+		Required: true,
+		MinItems: 1,
 	},
 	"expose_progress": IdentifierSchema(IdentifierSchemaParams{
 		Elem:        "expose_progress",
 		Description: "The name of the progress subsource for the source. If this is not specified, the subsource will be named `<src_name>_progress`.",
 		Required:    false,
 		ForceNew:    true,
 	}),
-	"subsource":      SubsourceSchema(),
```

Comment:
We should remove the `func SubsourceSchema()` from schema.go too, no?

Reply:
Good point, just removed that for the rest of the sources as well.